### PR TITLE
Referrals landing page

### PR DESCRIPTION
**Pull request by hedvigoscar**

### CODE REVIEW

File: src/pages/Referral/util.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+export const getFirebaseLinkDomain = () => {
+  if (
+    typeof window !== 'undefined' &&
+    (window as any).FIREBASE_LINK_DOMAIN !== undefined
+  ) {
+    return (window as any).FIREBASE_LINK_DOMAIN
```
Comment (palmenhq) [inline src/pages/Referral/util.ts]: we're making `return`s in virtually every branch here, do we really need all these `else`s n stuff? get's hard to read

File: src/routes.ts [hunk 2]
```diff
@@ -34,5 +35,6 @@ export const reactPageRoutes = [
     exact: true,
   },
   { path: '/dont-panic/hedvig', Component: LazyDontPanic, exact: true },
+  { path: '/referrals/:code', Component: Referral, exact: true },
```
Comment (palmenhq) [inline src/routes.ts]: we need to remember to add this to CF but awesome 🙌

File: src/pages/Referral/index.tsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+  marginLeft: 16,
+  marginRight: 16,
+})
+
+const Centered = styled('div')({
+  marginTop: 65,
+  textAlign: 'center',
+})
+
+const LinkButton = styled(Button.withComponent('a'))({
```
[the remark this answers is not in the document]
Comment (hedvigoscar) [reply src/pages/Referral/index.tsx]: ✅

File: src/pages/Referral/index.tsx [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+  fontWeight: 400,
+  padding: '16px 32px',
+})
+
+const PageContent: React.FunctionComponent<{ data: Data; code: string }> = ({
+  data: {
+    referralCampaignMemberInformation: {
+      incentive: { amount },
```
[the remark this answers is not in the document]
Comment (hedvigoscar) [reply src/pages/Referral/index.tsx]: We had multiple values inside the `referralCampaignMemberInformation`-object before, but yes, it makes no sense now. I've refactored it around a bit

File: src/pages/Referral/index.tsx [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+              return <div>Loading...</div>
+            }
+
+            return <PageContent data={data} code={code} />
+          }}
+        </Query>
+      )}
+    </Page>
+  ) : (
+    <Redirect to="/404" />
```
[the remark this answers is not in the document]
Comment (hedvigoscar) [reply src/pages/Referral/index.tsx]: How would that look? Something like:

```tsx
...
<Mount on={() => (window.location = 'https://www.hedvig.com/404')} />
...
```
?

File: src/pages/Referral/util.ts (hunk 1 above)
Comment (hedvigoscar) [reply src/pages/Referral/util.ts]: I tried to mirror the design found in `utils/apolloClient.ts::getGiraffeEndpoint`, but I'm open for suggestions on how to express it of course 🙂

File: src/routes.ts (hunk 2 above)
Comment (hedvigoscar) [reply src/routes.ts]: Absolutely! 🙂

File: src/pages/Referral/index.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (palmenhq) [reply src/pages/Referral/index.tsx]: yup look good. Maybe we could refactor it into a separate component to hide away the details of it

File: src/pages/Referral/util.ts (hunk 1 above)
Comment (palmenhq) [reply src/pages/Referral/util.ts]: Blaming that file i feel a bit silly hehe 😬 imo not one of my strongest pieces of code. imo it should be refactored into something like

```ts
  if (
    typeof window !== 'undefined' &&
    (window as any).FIREBASE_LINK_DOMAIN !== undefined
  ) {
    return (window as any).FIREBASE_LINK_DOMAIN
 }

  if (process.env.NODE_ENV === 'development') {
    return process.env.FIREBASE_LINK_DOMAIN || 'https://hedvigtest.page.link'
  } 

  if (
    process.env.NODE_ENV !== 'development' &&
    process.env.FIREBASE_LINK_DOMAIN
  ) {
    return process.env.FIREBASE_LINK_DOMAIN
  }

  throw Error(
    'Unable to find firebase link domain - specify with envvar `FIREBASE_LINK_DOMAIN`',
  )
```

File: src/pages/Referral/index.tsx [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+      >
+        <TranslationsConsumer textKey="REFERRAL_LANDINGPAGE_BTN_CTA">
+          {(text) => text}
+        </TranslationsConsumer>
+      </LinkButton>
+    </Centered>
+  </>
+)
+
+const MOCKED = true
```
Comment (palmenhq) [inline src/pages/Referral/index.tsx]: this feels a bit suspicious also hehe

Comment (hedvigoscar) [reply src/pages/Referral/index.tsx]: Interestingly enough it's going away completely - this view will no longer make API calls to giraffe.

So yeah, basically the entire reason I built in in web-onboarding has now gone away 😞

Comment (palmenhq) [reply src/pages/Referral/index.tsx]: haha good 😅

File: src/pages/Referral/util.ts (hunk 1 above)
Comment (hedvigoscar) [reply src/pages/Referral/util.ts]: Far prettier - I'm a huge fan of the early return over else's as well 😉

File: src/pages/Referral/index.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (hedvigoscar) [reply src/pages/Referral/index.tsx]: Looks like the path never even matches without the parameter, so I'm just going to remove the code altogether

File: src/pages/Referral/index.tsx [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+import { RouteComponentProps } from 'react-router'
+import { MobileContext } from 'utils/mobileContext'
+import { getFirebaseLinkDomain } from './util'
+
+type ReferralProps = RouteComponentProps<{
+  code: string
+}>
+
+const PageWrapper = styled('div')({
+  ['@media (min-width: 700px)']: {
```
Comment (palmenhq) [inline src/pages/Referral/index.tsx]: no need for square brackets here, does prettier not remove this?

Comment (hedvigoscar) [reply src/pages/Referral/index.tsx]: Nope 🤔 But I'll remove it!

Comment (palmenhq) [reply src/pages/Referral/index.tsx]: haha good

**Pull request merged**